src/app/payment/paymentMethod/DigitalRiverPaymentMethod.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import { default as PaymentMethodComponent, PaymentMethodProps } from './PaymentMethod';
+import PaymentMethodId from './PaymentMethodId';
+
+describe('when using Digital River payment', () => {
+    let method: PaymentMethod;
+    let checkoutService: CheckoutService;
+    let checkoutState: CheckoutSelectors;
+    let defaultProps: PaymentMethodProps;
+    let localeContext: LocaleContextType;
+    let PaymentMethodTest: FunctionComponent<PaymentMethodProps>;
+
+    beforeEach(() => {
+        defaultProps = {
+            method: getPaymentMethod(),
+            onUnhandledError: jest.fn(),
+        };
+
+        checkoutService = createCheckoutService();
+        checkoutState = checkoutService.getState();
+        localeContext = createLocaleContext(getStoreConfig());
+        method = { ...getPaymentMethod(), id: PaymentMethodId.DigitalRiver };
+
+        jest.spyOn(checkoutState.data, 'getConfig')
+            .mockReturnValue(getStoreConfig());
+
+        jest.spyOn(checkoutService, 'deinitializePayment')
+            .mockResolvedValue(checkoutState);
+
+        jest.spyOn(checkoutService, 'initializePayment')
+            .mockResolvedValue(checkoutState);
+
+        PaymentMethodTest = props => (
+            <CheckoutProvider checkoutService={ checkoutService }>
+                <LocaleContext.Provider value={ localeContext }>
+                    <Formik
+                        initialValues={ {} }
+                        onSubmit={ noop }
+                    >
+                        <PaymentMethodComponent { ...props } />
+                    </Formik>
+                </LocaleContext.Provider>
+            </CheckoutProvider>
+        );
+    });
+
+    it('renders as hosted widget method', () => {
+        const container = mount(<PaymentMethodTest { ...defaultProps } method={ method } />);
+        const component: ReactWrapper<HostedWidgetPaymentMethodProps> = container.find(HostedWidgetPaymentMethod);
+
+        expect(component.props())
+            .toEqual(expect.objectContaining({
+                containerId: 'drop-in',
+                deinitializePayment: expect.any(Function),
+                initializePayment: expect.any(Function),
+                method,
+            }));
+    });
+
+    it('initializes method with required config', () => {
+        const container = mount(<PaymentMethodTest { ...defaultProps } method={ method } />);
+        const component: ReactWrapper<HostedWidgetPaymentMethodProps> = container.find(HostedWidgetPaymentMethod);
+
+        component.prop('initializePayment')({
+            methodId: method.id,
+            gatewayId: method.gateway,
+        });
+
+        expect(checkoutService.initializePayment)
+            .toHaveBeenCalledWith(expect.objectContaining({
+                methodId: method.id,
+                gatewayId: method.gateway,
+                digitalriver: expect.objectContaining({
```

Comment:
it would be better if we can validate not only the type but the value, and make the test more reliable about the hardcoded values i.e. showSavePaymentAgreement, showComplianceSection, button, etc.

In case of callbacks you can keep the expect.any(Function).